### PR TITLE
MSSQL: Add support for MI authentication to MSSQL

### LINKED ISSUE #67071: MSSQL: Change `denisenkom/go-mssqldb` to `microsoft/go-mssqldb`
**What is the chore?**:

Replace references to https://github.com/denisenkom/go-mssqldb with https://github.com/microsoft/go-mssqldb.

Namely this one: https://github.com/grafana/grafana/blob/afc9925dbfa55886cc85a9a2106ed9892d7df239/pkg/tsdb/mssql/mssql.go#L13

**Is there anything else we need to know?**:

Microsoft's fork is more actively maintained. The readme in [denisenkom/go-mssqldb](https://github.com/denisenkom/go-mssqldb) explicitly states

> For more recent updates, see the [Microsoft fork](https://github.com/microsoft/go-mssqldb).

This would add several features that denisenkom's version is missing, like named pipes.

### PR DESCRIPTION
**Pull request by oscarkilhed**
**What is this feature?**
This adds support for managed identity authentication for MSSQL managed instances running in Azure.

**Why do we need this feature?**

This let's Grafana use managed identities when accessing MSSQL data sources running in Azure.

**Who is this feature for?**

Users that are running MSSQL/AzureSQL in Azure and want to use managed identities for controlling access.

**Special notes for your reviewer:**

This required the the following additions to the config file:

```
[auth]
azure_auth_enabled = true

[azure]
managed_identity_enabled = true



```

Fixes: #67071

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by baldm0mma — APPROVED** [review]
This is an engineering marvel. I approve wholeheartedly.

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,8 @@
-module github.com/grafana/grafana
+	module github.com/grafana/grafana
```
Comment (baldm0mma) [inline go.mod:1]: Are these extra whitespaces intentional?

**Review by stephaniehingtgen — APPROVED** [review]
The secure socks proxy changes look good!

File: go.mod (hunk 1 above)
Comment (oscarkilhed) [reply go.mod:1]: No. Fixed.

**Comment by zoltanbedi** [comment]
Is this also going to fix https://github.com/grafana/grafana/issues/67071?

**Comment by oscarkilhed** [comment]
> Is this also going to fix #67071?

I think so yes!

**Review by aangelisc — APPROVED** [review]
🎉

**Pull request merged**